applications/asset_tracker/src/light_sensor/light_sensor.c

```diff
@@ -105,6 +121,30 @@ void light_sensor_poll_fn(struct k_work *work)
 		ls_cb();
 	}
 
-	k_delayed_work_submit(
-		&ls_poller, K_SECONDS(CONFIG_LIGHT_SENSOR_DATA_SEND_INTERVAL));
+	submit_poll_work(data_send_interval_s);
+}
+
+void light_sensor_set_send_interval(const s32_t interval_s)
+{
+	if (interval_s == data_send_interval_s) {
+		return;
+	}
+
+	data_send_interval_s = interval_s;
+
+	if (!initialized) {
+		return;
+	}
+
+	if (data_send_interval_s) {
+		/* restart work for new interval to take effect */
+		submit_poll_work(K_NO_WAIT);
```

Comment:
If we `restart work for new interval`, why is it `K_NO_WAIT` then?

Reply:
this was a simple way to handle the interval change, instead of calculating a delay based on the time remaining and new interval.
a new reading is taken immediately and then the delayed wait will occur using the new interval.